functions-introduction/solutions/weather-quarkus/src/main/java/functions/Methods.java

```diff
@@ -0,0 +1,49 @@
+package functions;
+
+import java.io.*;
+import java.net.URISyntaxException;
+import java.net.URL;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+public class Methods
+{
+    ClassLoader classLoader = getClass().getClassLoader();
+    String jsonFileName = "cities.json";
+    Object weather = new Object();
+    File jsonResourceFile;
+
+    public Object read_weather() throws URISyntaxException, FileNotFoundException, IOException, ParseException
```

Comment:
I don't fully understand the logic behind this method. At this moment it does:
- Generates an input stream from a file name
- Checks if the input stream is  null and throws an exception in that case
- Generates a resource
- Checks if the resource is null and throws an exception in that case
- Finally we get the content of the file and return it as an object

I don't understand why are we creating and checking the input stream if we are not using to do anything.
Also why are we using the classLoader and URL things instead of using direct file read?  An example of a simple use Json with FileReader https://howtodoinjava.com/java/library/json-simple-read-write-json-examples/  and an example of a Quarkus project using FileReader to read files https://github.com/tnscorcoran/harry-potter-quarkus/blob/master/src/main/java/org/infinispan/hp/service/DataLoader.java

There is some room for improvements also here because there is too much indentation (generated by the conditionals) that makes the code complex and difficult to understand/read. You might want to use guard clauses (https://refactoring.guru/es/replace-nested-conditional-with-guard-clauses)

Reply:
The "cities.json" file is located under the resources folder. And to read/write content from a file present under the same gets implemented in a similar fashion.

Initially, I tried to shorten the approach by following alternate ways, but things didn't work out!!